packages/x-components/src/x-modules/related-tags/components/related-tag.vue

```diff
@@ -53,22 +50,21 @@
     public selectedRelatedTags!: RelatedTagModel[];
 
     /**
-     * Events list which is going to be emitted when a related tag is selected.
+     * Emits events when the button is clicked.
      *
-     * @returns The {@link XEvent | XEvents} to emit.
-     *
-     * @internal
+     * @public
      */
-    protected get events(): Partial<RelatedTagsXEvents> {
-      return this.isSelected
-        ? {
-            UserPickedARelatedTag: this.relatedTag,
-            UserDeselectedARelatedTag: this.relatedTag
-          }
-        : {
-            UserPickedARelatedTag: this.relatedTag,
-            UserSelectedARelatedTag: this.relatedTag
-          };
+    protected emitEvents(): void {
+      this.$x.emit('UserPickedARelatedTag', this.relatedTag, {
+        target: this.$el as HTMLElement
+      });
+      this.$x.emit(
```

Comment:
This is making the test fail. Because the wires of the `UserPickedARelatedTag` are synchronous, when the `UserDeselectedARelatedTag` or `UserSelectedARelatedTag` events are emitted, the related tag already has its new value, so you no longer need to flip the condition.

As emitting this events from here is a ñapa, I would say you can pick any solution, but leaving a friendly comment:

- Flip the events
- Emit the selected/deselected event before the picked one